VerificationWeb/Services/RoleService.cs

```diff
@@ -10,25 +10,65 @@ namespace VerificationWeb.Services
 {
     public class RoleService
     {
-        private readonly SocketGuild _guild;
+        private readonly DiscordSocketClient _client;
         public readonly Config Config;
 
         public RoleService(Config config, DiscordSocketClient client)
         {
             Config = config;
-            _guild = client.Guilds.FirstOrDefault(x => x.Id == Config.GuildId);
+            _client = client;
         }
 
-        public async Task AssignRoleAsync(ulong userId, IEnumerable<string> roleNames)
+        // Returns name of the roles
+        public async Task<List<string>> AssignRoleAsync(ulong userId, string groups)
         {
+            var roleNames = new List<string>();
             var rolesToAdd = new List<IRole>();
-                foreach (string roleName in roleNames)
+
+            foreach (var guild in _client.Guilds)
+            {
+                
+                if (!guild.Users.Any(x => x.Id == userId))
+                    break;
```

Comment:
> `guild.Users.Any`

`GetUser` by ID is o1 operation (followed by user == null then skip guild) while this^ is an iteration through possibly tens of thousands of entries ;)

> `break;`

`continue;` ?